test/Test/Database/LSMTree/Internal/Entry.hs

```diff
@@ -47,16 +54,32 @@ prop_resolveEntriesNormalSemantics ::
   => NonEmpty (Normal.Update v b)
   -> Property
 prop_resolveEntriesNormalSemantics es = expected === real
-  where expected = Just . unNormalUpdateSG . sconcat . fmap NormalUpdateSG $ es
-        real     = entryToUpdateNormal (sconcat (fmap updateToEntryNormal es))
+  where
+    expected = from . sconcat . to $ es
+      where
+        to = fmap NormalUpdateSG
+        from = Just . unNormalUpdateSG
+
+    real = from . sconcat . to $ es
+      where
+        to = fmap (EntrySG . updateToEntryNormal)
+        from (EntrySG x) = entryToUpdateNormal x
```

Comment:
Small nit, feel free to ignore: The two definitions looking that similar makes it a little unclear what the difference is, although I can see it when looking at each `to`. Maybe it would be clearer if called `toUpdate` and `toEntry`, or explicitly having `sconcat @EntrySG` etc.?

Reply:
I've added some type signatures to make it more clear